stubs/redis/redis/lock.pyi

```diff
@@ -0,0 +1,49 @@
+from types import TracebackType
+from typing import Optional, Text, Type, Union
+
+from redis.client import Redis
+
+_TokenValue = Union[bytes, Text]
+
+class Lock:
+    def __init__(
+        self,
+        redis: Redis,
+        name: str,
+        # could be Union[int, float]
+        timeout: Union[None, int, float] = ...,
```

Comment:
If the default is None, it's correct to use Optional (or Union[None, ...], which is equivalent).